File: ass47/program2.c
/*
2. Write a recursive program which accept number from user and return largest digit 
Input : 87983 
Output : 9 
*/
#include<stdio.h>
int Display(int x)
{
    int remider = 0;
    static int Largest = 0;
    if(x!=0)
    {   
        remider = x % 10;
        if(Largest<remider)
        {
            Largest = remider;
        }

        Display(x/10);
    }
    return Largest;
}

int main()
{
    int iNo = 0,iRet = 0;
    printf("Enter the Element\n");
    scanf("%d",&iNo);

    iRet = Display(iNo);
   printf("%d",iRet);
    return 0;
}

File: ass68/program2.java
/*
2. Write java program which accept array from user and replace each member with summation of its digit. 
Input :  89 687 56 549 87 9 
Output :  17 21 11 18 15 9
*/
import java.lang.*;
import java.util.*;
class MyArray
{
    void Display(int arr[])
    {
        for(int i=0; i<arr.length; i++)
        {
            int add = 0;
            while(arr[i]!=0)
            {
            add = add +arr[i]%10;
            arr[i]=arr[i]/10;
            }
            System.out.print(add+" ");
        }
    }
}

class Program2
{
	public static void main(String arg[])
	{
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number of first array");
		int no1 = sobj.nextInt();

        int arr[] = new int[no1];

		System.out.println("Enter the first array");

		for( int i = 0; i<arr.length; i++)
		{
			arr[i]= sobj.nextInt();
        }
        
        MyArray dobj = new MyArray();
        dobj.Display(arr);

        
	}
}